cmd/snap-confine/snap-confine.c

```diff
@@ -648,14 +658,21 @@ static bool is_device_cgroup_self_managed(const sc_invocation *inv)
 		die("cannot open %s", info_path);
 	}
 
+	sc_error *err SC_CLEANUP(sc_cleanup_error) = NULL;
 	char *self_managed_value SC_CLEANUP(sc_cleanup_string) = NULL;
-	sc_error *err = NULL;
 	if (sc_infofile_get_key
 	    (stream, "self-managed", &self_managed_value, &err) < 0) {
 		sc_die_on_error(err);
 	}
+	rewind(stream);
+
+	char *devmode_value SC_CLEANUP(sc_cleanup_string) = NULL;
+	if (sc_infofile_get_key(stream, "devmode", &devmode_value, &err) < 0) {
+		sc_die_on_error(err);
```

Comment:
will there be instances where the conffile exists but without this new devmode value and hence will stop the snap from launching? ie. should this just assume devmode is false if it is missing?

Reply:
In the case key is missing `devmode_value` will be set to NULL, which then fails `st_streq(.., "true")` check. `sc_infofile_get_key` only returns errors if the file is malformed or when passed incorrect input arguments. We already rely on this behavior, as `devmode=true` is only added if the snap is indeed installed in devmode.

Reply:
ah ok - thanks for clarifying that for me.